### PR TITLE
958 feat add link type to cells on table and smart table

### LINKED ISSUE #958: feat: add link type to cells on table and smart-table
Should be possible to add links to cells on a column.

image

### PR DESCRIPTION
**Pull request by allan-chagas-brisa**
## Issue Number

fix #958 

## Description

Changes on the link component usability and the addition of it to the table and smart table cells.

## Proposed Changes

- Added another anchor element to the link component without the href attr, so it wont affect the link with a click action;
- Added another type to the table colmuns interface;
- Added the link element to smart table and table cell types area;
- Added a "link config" to the table columns type, so it can universally pass to all links the same config, and also pass functions to set the link label, the condition to disable it, the action to execute on click or the link that the component refers to.

## Screenshots

image

## View Storybook

[Storybook](https://62eab350a45bdb0a5818520e-wcutrmcabm.chromatic.com/)

## Compliance

- [x] I have verified that this change complies with our code and contribution policies.
- [x] I have verified that this change does not cause regressions and does not affect other parts of the code.

### CODE REVIEW

**Review by iurynogueira — APPROVED** [review]
Awesome!

File: projects/ion/src/lib/link/link.component.html [hunk 1]
```diff
@@ -1,13 +1,12 @@
-<span [class.link-container]="disabled">
+<span class="link-container" [class.link-container--disabled]="disabled">
   <a
+    *ngIf="!link"
     data-testid="ion-link"
     class="ion-link"
     [class.ion-link--left-icon]="iconSide === 'left'"
     [class.ion-link--icon-only]="!label.length"
     [attr.disabled]="disabled"
     [target]="target"
```
Comment (danilo-moreira-brisa) [inline projects/ion/src/lib/link/link.component.html:9]:
```suggestion
    
```
You can remove this when there is no link.

File: projects/ion/src/lib/link/link.component.html [hunk 2]
```diff
@@ -24,4 +23,30 @@
       [size]="iconSize"
     ></ion-icon>
   </a>
+
+  <a
+    *ngIf="link"
+    data-testid="ion-link"
+    class="ion-link"
+    [class.ion-link--left-icon]="iconSide === 'left'"
+    [class.ion-link--icon-only]="!label.length"
```
Comment (danilo-moreira-brisa) [inline projects/ion/src/lib/link/link.component.html:32]:
```suggestion
    [class.ion-link--icon-only]="!label || label === ''"
```

File: projects/ion/src/lib/smart-table/smart-table.component.html [hunk 3]
```diff
@@ -166,6 +166,18 @@
                   row[column.tag.tooltipKey] || row[column.key]
                 "
               ></ion-tag>
+              <ion-link
+                *ngIf="column.type === 'link'"
```
Comment (danilo-moreira-brisa) [inline projects/ion/src/lib/smart-table/smart-table.component.html:170]:
```suggestion
                *ngIf="column.type === 'link' && column.link"
```

File: projects/ion/src/lib/table/utilsTable.ts [hunk 4]
```diff
@@ -28,6 +36,18 @@ interface TagRow {
...
+interface LinkRow {
+  label?: (_: SafeAny) => string;
+  icon?: IconType;
+  iconSide?: IconSide;
+  size?: FontSize;
+  bold?: boolean;
+  disabled?: (_: SafeAny) => boolean;
+  target?: LinkTarget;
+  url?: (_: SafeAny) => string;
+  action?: (_: SafeAny) => void;
```
Comment (danilo-moreira-brisa) [inline projects/ion/src/lib/table/utilsTable.ts:48]: All those SafeAny should be the same type as row, right?

**Review by danilo-moreira-brisa — CHANGES_REQUESTED** [review]

**Review by danilo-moreira-brisa — APPROVED** [review]

**Review by pedro-martins-brisa — APPROVED** [review]

**Pull request merged**